database/transaction.go

```diff
@@ -128,6 +128,49 @@ func (tx *Transaction) GetTable(name string) (*Table, error) {
 	}, nil
 }
 
+// modifyTable modifies TableInfo using given callback.
+func (tx *Transaction) modifyTable(name string, f func(*TableInfo) error) error {
```

Comment:
This should be a method on `tableInfoStore` because we want to do that update atomically, using the provided mutex.
The problem here is that we can have race conditions between the call to `tableInfoStore.Get`, `tableInfoStore.Delete` and `tableInfoStore.Insert`.